mlflow/utils/async_logging/async_logging_queue.py

```diff
@@ -0,0 +1,192 @@
+        except Exception as e:
+            _logger.error(f"Error while callback from atexit in _at_exit_callback: {e}")
+
+    def _logging_loop(self) -> None:
+        """
+        Continuously processes run data from the logging queue until `continue_to_process_data`
+         is False.
+        If an exception is raised during processing, it is logged and re-raised.
+        """
+        try:
+            while self.continue_to_process_data:
+                self._log_run_data()
+        except Exception as e:
+            from mlflow.exceptions import MlflowException
+
+            raise MlflowException(f"Exception inside the run data logging thread: {e}")
+
+    def _log_run_data(self) -> None:
+        """
+        Process the run data in the running runs queues.
+
+        For each run in the running runs queues, this method retrieves the next batch of run data
+         from the queue and processes it by calling the `_processing_func` method with the run ID,
+           metrics, parameters, and tags in the batch.
+        If the batch is empty, it is skipped. After processing the batch, the processed watermark
+         is updated and the batch event is set.
+        If an exception occurs during processing, the exception is logged and the batch event is set
+        with the exception. If the queue is empty, it is ignored.
+
+        Returns: None
+        """
+        run_batch = None  # type: RunBatch
+        for run_id, run_queue in self._active_run_data_batches.items():
+            while not run_queue.empty():
+                try:
+                    run_batch = run_queue.get(timeout=1)
+                except Empty:
+                    # Ignore empty queue exception
+                    continue
+                try:
+                    self._logging_func(
+                        run_id=run_id,
+                        metrics=run_batch.metrics,
+                        params=run_batch.params,
+                        tags=run_batch.tags,
+                    )
+
+                    # Signal the batch processing is done.
+                    run_batch.completion_event.set()
+
+                except Exception as e:
+                    _logger.error(f"Run Id {run_id}: Failed to log run data: Exception: {e}")
+                    run_batch.exception = e
+                    run_batch.completion_event.set()
+
+    def _wait_for_batch(self, batch: RunBatch) -> None:
+        """
+        Wait for the given batch to be processed by the logging thread.
+
+        Args:
+            batch: The batch to wait for.
+
+        Raises:
+            Exception: If an exception occurred while processing the batch.
+        """
+        try:
+            batch.completion_event.wait()
+            if batch.exception:
+                raise batch.exception
+        except Exception as e:
+            _logger.error(f"{batch.run_id}: Exception while waiting for batch: {e}")
```

Comment:
this error message will provide things like "23424tsdfq31: Exception while waiting for batch: runtimeError: SIGTERM ....", which is a bit weird to read.  

Actually I think we can remove this `try-except` block, we are just doing a wait() here without extra logic. The logging exception has already been handled.